apps/pipeline-visualization/src/mermaid_utils.ts

```diff
@@ -41,36 +40,90 @@ function escapeHtml(unsafe: string) {
     .replace(/'/g, '&#039;');
 }
 
-export function createMermaidPipeline(
+export function processMermaidLinks(
   pipeline: PipelineDefinition,
   mermaidOptions: MermaidOptions,
 ) {
-  const name: string = 'subgraph ' + pipeline.name;
-  const direction: string = 'direction ' + subgraphDirection;
-  const listOfPipes: Array<string[]> = [];
-  const listOfBocks: Array<string> = [];
-  let composites = '';
-  const process_pipe = (pipe: string[], block: BlockDefinition) => {
-    pipe.push(block.name);
-    if (isCompositeBlocktypeDefinition(block.type.ref)) {
-      if (mermaidOptions.compositeBlocks) {
-        const compositePipe: string[] = [];
-        for (const subblock of block.type.ref.blocks) {
-          compositePipe.push(subblock.name);
-        }
-        const compositeName = 'subgraph ' + block.name;
-        composites +=
-          compositeName +
-          '\n' +
-          direction +
-          '\n' +
-          compositePipe.join('-->') +
-          '\n' +
-          'end \n';
+  const processBlock = (block: BlockDefinition, depth = 0): string => {
+    const chain: string[] = [block.name];
+    const compositeBlocks = [];
```

Comment:
It's recommended to define the type explicitly when initializing a variable with an empty array. Just like you did in the row before.
I.e. `const compositeBlocks: MyType[] = [];`